### DISCUSSION TITLE
Pyinstaller, I can't see the icon on taskbar.

### DISCUSSION THREAD

**Discussion by Ardasak** [pyinstaller/pyinstaller#6084] [Help]
I'm trying to put an icon to my app, when I run the code `pyinstaller --noconsole --onefile --icon="icon.ico" main.py` and create the executable file, I can see it as the icon of exe file (photo_1) but after I executed it, I can't see the icon on taskbar (photo_2).

[photo_1](https://i.stack.imgur.com/yqR0Z.png)

[photo_2](https://i.stack.imgur.com/5kgiO.png)

I'd like to learn why is this happening and how to fix it, thanks for your help.

**Comment by bwoodsend** [comment]
The taskbar icon isn't controlled by PyInstaller. You need to set it in Python code using whichever GUI framework that you are using. You'll have to tell me which framework if you want an example.

**Reply by Ardasak** [reply]
I'm using PyQt5

**Reply by bwoodsend** [reply]
```python
# test.py
"""A minimal PyQt5 application with a taskbar icon."""

import sys
from pathlib import Path

from PyQt5.QtWidgets import QWidget, QApplication
from PyQt5.QtGui import QIcon


class MainWindow(QWidget):
    # Pretent there's something interesting here.
    pass


qapp = QApplication(sys.argv)

# --- Magic happens here ---
# Get the icon's filename.
icon_path = Path(__file__).with_name("icon.png")
# Open the icon and set the Window/taskbar icon.
qapp.setWindowIcon(QIcon(str(icon_path)))

# Then the usual Qt boiler plate...
window = MainWindow()
window.show()
qapp.exec()
```
I'm using this [icon.png](https://user-images.githubusercontent.com/30940778/127883668-c4a69100-44a0-4f49-908c-6e046a394eb4.png) in the same folder as `test.py`. Note that it must be a PNG - PyQ5 will ignore your .ico file so you will have to convert it. 

On Windows build with:
```
pyinstaller --add-data=icon.png;. test.py
```
On anything else use:
```
pyinstaller --add-data=icon.png:. test.py
```

**Reply by egeakman** [reply]
> ```python
> # test.py
> """A minimal PyQt5 application with a taskbar icon."""
> 
> import sys
> from pathlib import Path
> 
> from PyQt5.QtWidgets import QWidget, QApplication
> from PyQt5.QtGui import QIcon
> 
> 
> class MainWindow(QWidget):
>     # Pretent there's something interesting here.
>     pass
> 
> 
> qapp = QApplication(sys.argv)
> 
> # --- Magic happens here ---
> # Get the icon's filename.
> icon_path = Path(__file__).with_name("icon.png")
> # Open the icon and set the Window/taskbar icon.
> qapp.setWindowIcon(QIcon(str(icon_path)))
> 
> # Then the usual Qt boiler plate...
> window = MainWindow()
> window.show()
> qapp.exec()
> ```
> I'm using this [icon.png](https://user-images.githubusercontent.com/30940778/127883668-c4a69100-44a0-4f49-908c-6e046a394eb4.png) in the same folder as `test.py`. Note that it must be a PNG - PyQ5 will ignore your .ico file so you will have to convert it. 
> 
> On Windows build with:
> ```
> pyinstaller --add-data=icon.png;. test.py
> ```
> On anything else use:
> ```
> pyinstaller --add-data=icon.png:. test.py
> ```
> 

Thanks for your help, have a good day.

**Reply by ghost** [reply]
Hey, I have the same issue only I am using pygame. I did use pygame.display.set_icon and it shows up on the window frame and on the exe, just not on the taskbar. Weirdly it does show up correctly when the app is pinned to the taskbar. I have never had this issue before.
